cupy/testing/_array.py

```diff
@@ -92,18 +117,31 @@ def assert_array_equal(x, y, err_msg='', verbose=True, strides_check=False,
 
     .. seealso:: :func:`numpy.testing.assert_array_equal`
     """
-    numpy.testing.assert_array_equal(
-        cupy.asnumpy(x), cupy.asnumpy(y), err_msg=err_msg,
-        verbose=verbose, **kwargs)
+    if numpy.lib.NumpyVersion(numpy.__version__) >= '2.0.0':
```

Comment:
This is new in NumPy v1.24.
https://numpy.org/doc/1.24/reference/generated/numpy.testing.assert_array_equal.html